pkg/kubernetes/backup_storage.go

```diff
@@ -82,34 +83,49 @@ func (k *Kubernetes) IsBackupStorageUsed(ctx context.Context, namespace, name st
 	}
 
 	// Check if it is in use by clusters?
-	clusters, err := k.client.ListDatabaseClusters(ctx, namespace, metav1.ListOptions{})
-	if err != nil {
+	listOpts := &ctrlclient.ListOptions{
```

Comment:
this is an example of FieldSelector usage over `.spec.backup.schedules.backupStorageName` in DatabaseCluster CR